pkg/controller/controller.go

```diff
@@ -317,6 +381,46 @@ func (p *csiProvisioner) Provision(options controller.VolumeOptions) (*v1.Persis
 			RequiredBytes: int64(volSizeBytes),
 		},
 	}
+
+	if checkSnapshotSupport {
```

Comment:
maybe to put this part into a separate function to get snapshot handler?

Reply:
Done